### PR TITLE
Add missing report field to Notification entity

### LINKED ISSUE #21: Add missing fields to Notification entity
Mastodon provides a Notifications API which has API calls that return a `Notification` entity. While we already provide a basic implementation of this endpoint, the `Notification` entity currently does not implement all the fields returned by Mastodon servers. The following fields are missing and should be added to the entity:

- report

The Notification entity is documented here:
https://docs.joinmastodon.org/entities/Notification/

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
Closes #21 

# Description

Adds the missing `report` field to the `Notification` entity.

As I wasn’t sure where you usually get your test data from, I copied some parts from [Report](https://docs.joinmastodon.org/entities/Report/) and [Notification](https://docs.joinmastodon.org/entities/Notification/).

# Known issues 

When #20 is implemented, `actionTaken` should be changed to type `Boolean` in addition to adding the other fields.

It is currently coerced into a String by GSON but that might change with #224.

### CODE REVIEW

**Comment by andregasser** [comment]
Regarding your question where we get our test data from: Either from real instance API responses (data should then be anonymized) or from the API documentation example responses. It varies from time to time. Also, we sometimes also re-use existing JSON files (as you did in this case).

File: bigbone/src/main/kotlin/social/bigbone/api/entity/Report.kt [hunk 1]
```diff
@@ -17,5 +17,5 @@ data class Report(
      * Whether an action was taken yet.
      */
     @SerializedName("action_taken")
-    val actionTaken: String = ""
+    val actionTaken: String = "" // todo: Change this to type Boolean
```
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/api/entity/Report.kt:20]: Open TODO comment: I'd suggest that we change type to `Boolean` as you suggested.

**Review by andregasser — CHANGES_REQUESTED** [review]
Hello @PattaFeuFeu 
Thanks for the PR :-) I have added a minor comment regarding the data type in Report.kt. I think it is safe to change this now, as we did not even publish a real version of BigBone yet. Any objections from your side?

File: bigbone/src/main/kotlin/social/bigbone/api/entity/Report.kt (hunk 1 above)
Comment (PattaFeuFeu) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/Report.kt:20]: This is already changed in #277 but I can change it to Boolean here to reduce merge conflicts. 👍

Comment (andregasser) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/Report.kt]: No worry, you can leave it as it is.

Comment (PattaFeuFeu) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/Report.kt]: Done 👍

Comment (andregasser) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/Report.kt]: Also good :-)

**Review by andregasser — APPROVED** [review]

**Review by andregasser — APPROVED** [review]

**Pull request merged**